### PR TITLE
Reject database names with path segments over 128 characters

### PR DESCRIPTION
**Pull request by rnewson**
COUCHDB-2821

### CODE REVIEW

File: src/chttpd.erl [hunk 1]
```diff
@@ -820,6 +820,10 @@ error_info({error, {illegal_database_name, Name}}) ->
     {400, <<"illegal_database_name">>, Message};
 error_info({_DocID,{illegal_docid,DocID}}) ->
     {400, <<"illegal_docid">>,DocID};
+error_info({error, {database_name_too_long, DbName}}) ->
+    {400, database_name_too_long,
```
Comment (eiri) [inline src/chttpd.erl]: Is the second tuple's element is intentionally an atom? It seems to be a binary at the rest of the responses.

Comment (kxepal) [inline src/chttpd.erl]: Suddenly, atoms here works as expected. Not sure why we use binaries for else.

**Comment by kxepal** [comment]
+1

**Comment by kxepal** [comment]
(there is something really wrong with couch_log)

File: src/chttpd.erl (hunk 1 above)
Comment (rnewson) [inline src/chttpd.erl]: I can change to a binary, old habits.

**Pull request merged**